pkg/kubectl/cmd/set/set_resources.go

```diff
@@ -230,7 +241,11 @@ func (o *ResourcesOptions) Run() error {
 
 		//no changes
 		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
-			allErrs = append(allErrs, fmt.Errorf("info: %s %q was not changed\n", info.Mapping.Resource, info.Name))
+			if isSelectedPrintMode(o.Output) {
```

Comment:
This is going to make viewing the objects harder, because "not changed" and "changed" look similar enough that it will decrease readability.

I also think we should be printing this on stderr with an `info:` prefix because it is not expected for the user invoking this:

```
fmt.Fprintf(o.Err, "info: %s %q was not changed\n", info.Mapping.Resource, info.Name)
```

Reply:
i think users expect this resources is using expectd image instead of image happen updated.
So i 'd more like keep print to `o.Out`.
@liggitt wdyt?

Reply:
@smarterclayton sorry about this, this you are right. scene:
when i upgrade my all app, this can quick help find some image not changed by collect error in bash.
and i add `error:` in prefix